tests/publiccloud/img_proof.pm

```diff
@@ -148,19 +147,20 @@ sub run {
 
     assert_script_run('rm -rf img_proof_results');
 
+    if (is_hardened() && !check_var('SCAP_REPORT', 'skip')) {
+        # Upload SCAP profile used by img-proof
+        my $url = "https://ftp.suse.com/pub/projects/security/oval/suse.linux.enterprise.15.xml.gz";
+        assert_script_run("curl --fail -LO $url");
+        upload_logs("suse.linux.enterprise.15.xml.gz");
+    }
+
     # fail, if at least one test failed
     if ($img_proof->{fail} > 0) {
         $instance->run_ssh_command(cmd => 'rpm -qa > /tmp/rpm_qa.txt', no_quote => 1);
         upload_logs('/tmp/rpm_qa.txt');
         $instance->run_ssh_command(cmd => 'sudo journalctl -b > /tmp/journalctl_b.txt', no_quote => 1);
         upload_logs('/tmp/journalctl_b.txt');
-    }
-
-    if (is_hardened() && !check_var('SCAP_REPORT', 'skip')) {
-        # Upload SCAP profile used by img-proof
-        my $url = "https://ftp.suse.com/pub/projects/security/oval/suse.linux.enterprise.15.xml.gz";
-        assert_script_run("curl --fail -LO $url");
-        upload_logs("suse.linux.enterprise.15.xml.gz");
+        die('img_proof failed so we need to fail module');
```

Comment:
There's no need to have a message with low-level details justifying it.  Just a statement of fact:

```suggestion
        die('img_proof failed');
```